internal/controller/multiclusterservice_controller.go

```diff
@@ -192,9 +321,43 @@ func (r *MultiClusterServiceReconciler) reconcileDelete(ctx context.Context, mcs
 	return ctrl.Result{}, nil
 }
 
+// requeueSveltosProfileForClusterSummary asserts that the requested object has Sveltos ClusterSummary
+// type and requeues its owner, which is either a Sveltos Profile or ClusterProfile object.
+func requeueSveltosProfileForClusterSummary(ctx context.Context, o client.Object) []reconcile.Request {
+	l := ctrl.LoggerFrom(ctx)
+	msg := "cannot queue request"
+
+	cs, ok := o.(*sveltosv1beta1.ClusterSummary)
+	if !ok {
+		l.Error(fmt.Errorf("request is not for a ClusterSummary object"), msg, "Name", o.GetName(), "Namespace", o.GetNamespace())
+		return []ctrl.Request{}
+	}
+
+	ownerRef, err := sveltosv1beta1.GetProfileOwnerReference(cs)
+	if err != nil {
+		l.Error(err, msg, "Name", o.GetName(), "Namespace", o.GetNamespace())
+		return []ctrl.Request{}
+	}
+
+	// The Profile/ClusterProfile object has the same name as its
+	// owner object which is either ManagedCluster or MultiClusterService.
+	req := client.ObjectKey{Name: ownerRef.Name}
+	if ownerRef.Kind == sveltosv1beta1.ProfileKind {
+		req.Namespace = o.GetNamespace()
+	}
```

Comment:
could you elaborate on the comment and the idea behind the code block? there is a statement that the owner of a `clustersummary` object is either an `mcl` (ns-scoped) or `mcs` (cluster-scope), so why do we contest the owner reference's kind against sveltos' `profile`? 


just in case: I am not sure whether the logger already has `name`/`namespace` values, if so, then the same keys might be confusing, so maybe something similar (not exactly those) to the `clustersummary-name`/`clustersummary-namespace` keys

Reply:
I think the comment on the function I made might be confusing. It should be:
```go
// requeueSveltosProfileForClusterSummary asserts that the requested object has Sveltos ClusterSummary
// type, fetches its owner (a Sveltos Profile or ClusterProfile object), and requeues the ManagedCluster 
// or MultiClusterService object that is referenced by the same namespace/name as the Sveltos Profile/ClusterProfile.
```
The ownership of these objects goes like this:
```
ClusterSummary -> is owned by Profile/ClusterProfile -> is owned by ManagedCluster/MultiClusterService
```
Where the ManagedCluster and MultiClusterService have the same reference (namespace/name) as the Profile and ClusterProfile objects that the own respectively. So what this function is doing is:

1. Making sure that the requested reference is for a ClusterSummary object.
2. Fetching the reference for its owner (Profile or ClusterProfile) and requeuing that reference.
3. Since this reference is the same as that of an `mcl` or `mcs` object, we end up running reconcile on the `mcl` or `mcs` object, which is what we want to fetch the latest status from their respective ClusterSummary objects.

> just in case: I am not sure whether the logger already has name/namespace values

I can check that. Thanks for pointing it out.

Reply:
Ok, now I get it, it was my inattention to the code that I missed the ownership chain. So I think everything is fine here, and I suggest amending the func comment or adding an extra comment within the func just in case (I believe your comment here is very well described the behavior)